### PR TITLE
DM-48013: Add PR checklist mentioning APDB versioning

### PR DESCRIPTION
**Pull request by andy-slac**

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Checklist
+
+- [ ] If changing `apdb.yaml`, it usually requires version number increment, check [schema versioning page](https://rubinobs.atlassian.net/wiki/external/OGUyOGM1OTY0MTRkNGFmNThkNWM4N2U1NGFmODJkNWY) for details.
```
Comment (kfindeisen) [inline .github/pull_request_template.md:3]: I think @ebellm said that changes to `apdb.yaml` also need to be synchronized with https://github.com/lsst/alert_packet?

Comment (andy-slac) [reply .github/pull_request_template.md:3]: We can certainly mention that too, please suggest wording for it.

**Review by JeremyMcCormick — CHANGES_REQUESTED** [review]
I'm going to put a review hold on this for the time being so it doesn't get inadvertently merged. I discussed with Andy today about making a more generic checklist and contribution guide and will take over development on this branch.

I have no problem if we eventually create a dedicated page for APDB considerations within sdm_schemas itself, too, e.g., incorporating information like that mentioned in https://github.com/lsst/sdm_schemas/pull/290#discussion_r1874124079. That could be on a separate branch/ticket from the more generic contribution guide and checklist.

I don't think we should link to Confluence pages. I would prefer if it was self-contained and the checklist pointed to pages within the sdm_schemas repository, only.

File: .github/pull_request_template.md (hunk 1 above)
Comment (JeremyMcCormick) [reply .github/pull_request_template.md:3]: > I think @ebellm said that changes to `apdb.yaml` also need to be synchronized with https://github.com/lsst/alert_packet?

I see that there are detailed instructions under https://github.com/lsst/alert_packet/tree/main?tab=readme-ov-file#adding-a-new-schema

We can provide a direct link to this section in the sdm_schemas docs.

**Comment by JeremyMcCormick** [comment]
I'm going to re-open a new PR so Andy can be a reviewer.

**Pull request closed without merging**